bundle.py

```diff
@@ -0,0 +1,194 @@
+    requirements = [r.split('#')[0].strip() for r in requirements if r]
+
+    toml = tomlkit.parse(original_toml)
+    toml['tool']['briefcase']['version'] = VERSION
+    toml['tool']['briefcase']['app'][APP]['requires'] = requirements + [
+        "pip",
+        "PySide2==5.14.2.2",
+    ]
+
+    print("patching pyroject.toml to version: ", VERSION)
+    print(
+        "patching pyroject.toml requirements to : \n",
+        "\n".join(toml['tool']['briefcase']['app'][APP]['requires']),
+    )
+
+    with open(PYPROJECT_TOML, 'w') as f:
+        f.write(tomlkit.dumps(toml))
+
+
+def patch_dmgbuild():
+    if not MACOS:
+        return
+    from dmgbuild import core
+
+    # will not be required after dmgbuild > v1.3.3
+    # see https://github.com/al45tair/dmgbuild/pull/18
+    with open(core.__file__, 'r') as f:
+        src = f.read()
+    if 'max(total_size / 1024' not in src:
+        return
+    with open(core.__file__, 'w') as f:
+        f.write(src.replace('max(total_size / 1024', 'max(total_size / 1000'))
+        print("patched dmgbuild.core")
+
+
+def add_site_packages_to_path():
+    # on mac, make sure the site-packages folder exists even before the user
+    # has pip installed, so it is in sys.path on the first run
+    # (otherwise, newly installed plugins will not be detected until restart)
+    if MACOS:
+        pkgs_dir = os.path.join(
+            APP_DIR,
+            'Contents',
+            'Resources',
+            'Support',
+            'lib',
+            f'python{sys.version_info.major}.{sys.version_info.minor}',
+            'site-packages',
+        )
+        os.makedirs(pkgs_dir)
+        print("created site-packages at", pkgs_dir)
+
+    # on windows, briefcase uses a _pth file to determine the sys.path at
+    # runtime.  https://docs.python.org/3/using/windows.html#finding-modules
+    # We update that file with the eventual location of pip site-packages
+    elif WINDOWS:
+        py = "".join(map(str, sys.version_info[:2]))
+        python_dir = os.path.join(BUILD_DIR, APP, 'src', 'python')
+        pth = os.path.join(python_dir, f'python{py}._pth')
+        with open(pth, "a") as f:
+            # Append 'hello' at the end of file
+            f.write(".\\\\Lib\\\\site-packages\n")
+        print("added bundled site-packages to", pth)
+
+        pkgs_dir = os.path.join(python_dir, 'Lib', 'site-packages')
+        os.makedirs(pkgs_dir)
+        print("created site-packages at", pkgs_dir)
+        with open(os.path.join(pkgs_dir, 'readme.txt'), 'w') as f:
+            f.write("this is where plugin packages will go")
+
+
+def patch_wxs():
+    # must run after briefcase create
+    fname = os.path.join(BUILD_DIR, APP, f'{APP}.wxs')
+
+    if os.path.exists(fname):
+        with open(fname, 'r') as f:
+            source = f.read()
+        with open(fname, 'w') as f:
+            f.write(source.replace('pythonw.exe', 'python.exe'))
+            print("patched pythonw.exe -> python.exe")
+
+
+def make_zip():
```

Comment:
out of curiosity, why are we creating and uploading zips of the artifact?

Reply:
that was to make the CI process easier when building and deploying on github actions.  Also not married to that if someone wants to tinker with getting the github workflow working across platforms.